### DISCUSSION TITLE
Options for Health Drain

### DISCUSSION THREAD

**Discussion by WalmartWhiteCat** [ppy/osu#25631] [Ideas] [answered]
This is a really, incredibly small detail, that I'm sure at least 95% of people won't give a damn about, but I can just see people talking about it, specifically those who started on Lazer instead of Stable.

Maybe in the options, just make it so you can switch between Stable drain and Lazer drain. It could be nice for people who are used to the original drain on Lazer and would remove the need to rely on the use of a mod to recreate the feeling of more lenient drain in a subpar fashion (Like lowering HP with Difficulty Adjust).

**Comment by peppy** [comment]
Please use difficulty adjust and adjust the HP difficulty. I would not call this "subpar" but the correct way to do so.

**Reply by GuardianGuy** [reply]
I started on lazer and have flip-flopped between stable and lazer a few times and stable drain with DA and lazer drain do not feel the same.  It feels like slider ends specifically drain way more with stable drain. Could the switch maybe be a new for-fun mod? I don't want to have to stay on an old version just to keep the drain that's more comfortable.  If this isn't something the dev team will ever add in, could you point me towards resources on making custom mods (and maybe a way to make one that doesn't require copying the standard ruleset or rebuilding lazer from source)? All I can find is custom mode stuff

Somewhat related, but I feel like lazer drain is more welcoming for new players. A lot of times in stable I'll fail a map and can't figure out why even with slowing down the replay, but that hasn't happened with lazer.

**Reply by peppy** [reply]
Discussing this with @smoogipoo IRL, it's quite possible that drain is too harsh because of mechanic differences on lazer (ie. sliders don't have a judgement at the end, which would historically give an HP boost). Said mechanics are potentially changing via https://github.com/ppy/osu/pull/25594 which may bring this back closer in line with expectations, so we'll have to keep a watch on this, and on feedback regarding HP drain.

**Reply by smoogipoo** [reply]
I'm a bit confused, can you clarify a few things for me? When you mention "lazer drain" are you talking about the old drain algorithm, or the new one that intends to mimic stable's drain?

For example, are you saying that the new lazer drain doesn't match stable? This question: _"feels like slider ends specifically drain way more with stable drain"_ could be interpreted as:
- Slider ends with the new drain algorithm drain way more than osu!stable.
- Slider ends with the new drain algorithm drain way more than the old lazer algorithm.

**Reply by GuardianGuy** [reply]
Sorry for the confusion.  I meant that the current drain system in lazer (the one that is basically ported from stable) feels exactly like the drain in stable.  Having tried both the current (matching stable) and previous (before making lazer's like stable) drain systems, I do not like the current system due it having the harsher slider ends and the hard to describe quirks (more of a gut feeling type of thing, just doesn't feel right) that cause the mystery fails I mentioned.

Stable health drain feels harsh, pre-matching-stable lazer drain feels good

**Reply by smoogipoo** [reply]
Thanks for the clarification. That eases my mind a bit.

**Comment by smoogipoo** [comment]
Kind of stuck between a rock and hard place here.

At the moment we really don't want to make it either an option (because it wouldn't be stored to replays) or a mod (because it feels excessive). The only reasonable call I can make at the moment is to continue using the new implementation going forward.

The call I would _like_ to make is to use the old lazer algorithm as default and gate the new one behind the classic mod. Unfortunately I can't make this call because the old lazer algorithm received a lot of criticism of being either too lenient or too harsh at times itself.

For now, this will definitely have to wait upon https://github.com/ppy/osu/pull/25594, because the placement of slider judgements are ever so slightly different compared to osu!stable which the algorithm was designed for.

**Comment by Jimmy5522** [comment]
Hello!

I think the problem here is that there was such a significant _change_ in the new HP drain system, not the system itself.
It got significantly harder. I got knocked down nearly a full star in difficulty that I was able to complete.

No matter what system is in place, people are going to complain. However, failing songs that I was able to complete months ago is extremely frustrating. I don't have a lot of time to practice playing Osu!, and I really just want to have fun playing the game. I don't have the time to practice just to get back to where I was.

Although I understand why the changes were implemented, that doesn't really change the fact that it feels terrible to be set back so far. I think that smaller changes in the future would help the more casual community keep up when increasing the difficulty across the board.

**Reply by 99simo** [reply]
I have to agree with the frustration here, i started on lazer specifically because i tried regular osu and as a bad player i just didnt understand why some maps i would fail at 70-75%. 
Problem is im still bad and i still dont understand the health system. Im sure there is a lot of history on why it is the way it is and the discussion could go on forever but the frustration of both failing maps that i could pass before and failing maps while seeying what i consider an high number remains.
To give an example, every map that has a slow section with stable health drain just leaves you at no health even if you miss no notes, and at that point even if you miss a couple its a fail , it makes no sense to me. Why am i getting punished for time passing regardless of performance, why is health drain even a thing.
There is probably a larger discussion to be had on what is a pass, where is the bar for a pass supposed to be. 
I know i could just difficulty adjust but it just feels like im not actually playing the game once i start modifying difficulty.

**Reply by smoogipoo** [reply]
I think it would be helpful to provide example beatmaps or perhaps replays (where possible...) where drain is too harsh.

**Comment by shiumano** [comment]
Is it possible to add a "use old HP drain" setting to the Classic Mod?

**Reply by smoogipoo** [reply, marked as answer]
This has been done in the latest release, and non-classic uses the new algorithm by default.
